hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -466,9 +467,18 @@ public Hashtable<String, String> getFilesystemProperties(
               client.getFileSystem());
 
       final Hashtable<String, String> parsedXmsProperties;
+      final AbfsRestOperation op;
 
-      final AbfsRestOperation op = client
-          .getFilesystemProperties(tracingContext);
+      if (getPrefixMode() == PrefixMode.BLOB) {
+        op = client.getContainerMetadata(tracingContext);
+        perfInfo.registerResult(op.getResult());
+
+        parsedXmsProperties = parseResponseHeadersToHashTable(op.getResult());
+        perfInfo.registerSuccess(true);
```

Comment:
perfInfo.registerResult(op.getResult()).registerSuccess(true); can be changed to a single line